### PR TITLE
Implement the 'Add relationship' modal, only front end [SCI-9671]

### PR DESCRIPTION
**Pull request by lasniscinote**
Jira ticket: [SCI-9671](https://scinote.atlassian.net/browse/SCI-9671)

### What was done
Implemented front-end for the add relationship modal

[SCI-9671]: https://scinote.atlassian.net/browse/SCI-9671?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/packs/vue/add_item_relationships.js
Comment (okriuchykhin) [inline app/javascript/packs/vue/add_item_relationships.js:1]: name it repository_item_relationships_modal.js

File: app/javascript/vue/item_relationships/AddItemRelationshipsModal.vue [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    </div>
+  </div>
+</template>
+
+<script>
+import SelectSearch from "../shared/select_search.vue"
+import ChecklistSelect from "../shared/checklist_select.vue"
+
+export default {
+  name: 'AddItemRelationshipsModal',
```
Comment (okriuchykhin) [inline app/javascript/vue/item_relationships/AddItemRelationshipsModal.vue:102]: call it RepositoryItemRelationshipsModal

File: app/javascript/vue/repository_item_sidebar/RepositoryItemSidebar.vue
Comment (okriuchykhin) [inline app/javascript/vue/repository_item_sidebar/RepositoryItemSidebar.vue:1]: Remove autoformatting from here

File: app/javascript/vue/shared/checklist_select.vue [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <template>
   <div class="w-full relative" ref="container" v-click-outside="{ handler: 'close', exclude: ['optionsContainer'] }">
     <button ref="focusElement"
-            class="btn flex justify-between items-center w-full outline-none border-[1px] bg-white rounded p-2
-                  font-inter text-sm font-normal leading-5"
+            class="btn flex justify-between items-center w-full outline-none border-[1px] bg-white rounded p-2 pl-4
```
Comment (okriuchykhin) [inline app/javascript/vue/shared/checklist_select.vue:4]: is it a global change? Won't it affect other parts?

File: app/javascript/vue/item_relationships/RepositoryItemRelationshipsModal.vue [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      selectedItemValues: [],
+      selectedRelationshipValue: null,
+      inventoryOptions: [{ 0: 1, 1: 'Participants database' }, { 0: 2, 1: 'Inventory Option 2' }, { 0: 3, 1: 'Inventory Option 3' }, { 0: 4, 1: 'Inventory Option 4' }],
+      itemOptions: [{ id: 1, label: 'SEPA-STASE' }, { id: 2, label: 'GTC' }, { id: 3, label: 'DESTIL-MACRO' }, { id: 4, label: 'ARNESSLIM-L' }],
+      relationshipOptions: [{ 0: 1, 1: 'Parent' }, { 0: 2, 1: 'Child' }],
+    };
+  },
+  computed: {
+    shouldEnableAddButton() {
+      if ((this.selectedInventoryValue !== null) && (this.selectedRelationshipValue !== null) && (this.selectedItemValues.length > 0)) {
```
Comment (artoscinote) [inline app/javascript/vue/item_relationships/RepositoryItemRelationshipsModal.vue:128]: just return the expression

File: app/javascript/vue/item_relationships/RepositoryItemRelationshipsModal.vue [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      this.selectedRelationshipValue = value;
+    },
+    cancel() {
+      this.selectedInventoryValue = null;
+      this.selectedItemValues = [];
+      this.selectedRelationshipValue = null;
+      $(this.$refs.repositoryItemRelationshipsModal).modal('hide');
+      this.$delete(this.data, 'selectedInventoryValue');
+      this.$delete(this.data, 'selectedItemValues');
+      this.$delete(this.data, 'selectedRelationshipValue');
```
Comment (artoscinote) [inline app/javascript/vue/item_relationships/RepositoryItemRelationshipsModal.vue:163]: try and remove this delete part and test again

File: app/javascript/vue/repository_row/manage_stock_value_modal.vue [hunk 5]
```diff
@@ -63,6 +63,7 @@
                   :placeholder="i18n.t('repository_stock_values.manage_modal.unit_prompt')"
                   @change="unit = $event"
                   :className="`${errors.unit ? 'error' : ''}`"
+                  :shouldRecalcPosition="true"
```
Comment (artoscinote) [inline app/javascript/vue/repository_row/manage_stock_value_modal.vue:66]: remove changes to this file from this PR

**Review by artoscinote — CHANGES_REQUESTED** [review]

**Review by artoscinote — APPROVED** [review]

**Pull request merged**